src/psyclone/psyad/utils.py

```diff
@@ -42,39 +42,28 @@
 
 
 def node_is_active(node, active_variables):
-    ''' Determines whether this node contains variables that are active.
+    '''Determines whether this node contains variables that are active.
 
     :param node: the PSyIR node that is being evaluated.
     :type node: :py:class:`psyclone.psyir.nodes.Node`
-    :param active_variables: a list of active variables.
-    :type active_variables: list of \
-        :py:class:`psyclone.psyir.symbols.DataSymbol`
+    :param active_variables: a list of active variables either as \
+        PSyIR symbols or as strings.
+    :type active_variables: 
+        List[:py:class:`psyclone.psyir.symbols.DataSymbol`] or \
+        List[str]
 
     :returns: True if active and False otherwise.
     :rtype: bool
 
     '''
-    for reference in node.walk(Reference):
-        if reference.symbol in active_variables:
-            return True
-    return False
-
-
-def node_is_active_names(node, active_variable_names):
-    ''' Determines whether this node contains variables that are active.
-
-    :param node: the PSyIR node that is being evaluated.
-    :type node: :py:class:`psyclone.psyir.nodes.Node`
-    :param active_variable_names: a list of active variable names.
-    :type active_variable_names: list of str
-
-    :returns: True if active and False otherwise.
-    :rtype: bool
-
-    '''
-    for reference in node.walk(Reference):
-        if reference.symbol.name.lower() in active_variable_names:
-            return True
+    if active_variables and isinstance(active_variables[0], str):
+        for reference in node.walk(Reference):
+            if reference.symbol.name.lower() in active_variables:
+                return True
+    else:
+        for reference in node.walk(Reference):
+            if reference.symbol in active_variables:
+                return True
```

Comment:
Sorry to be a pain but I think we should be defensive here and raise an error if something other than a str or symbol is found. Probably we should do this for every element in the list - using an `all(isinstance(item, DataSymbol) for item in active_variables)` for DataSymbol and str.

Reply:
Please ensure you document the resulting :raises: too.

Reply:
Donerated including tests